celery/backends/mongodb.py

```diff
@@ -43,6 +46,33 @@ def __init__(self, **kw):
         self.__dict__.update(kw)
 
 
+def auto_retry(f):
+    ''' Forces decorated function to retry if mongo throws error '''
```

Comment:
Just for the mongo backend indeed. I merged in latest changes from master to resolve the conflicts